csrc/type.cpp

```diff
@@ -682,6 +682,12 @@ static const char* parallel_type2string(ParallelType t) {
       return "blockIdx.y";
     case ParallelType::BIDx:
       return "blockIdx.x";
+    case ParallelType::CIDz:
+      return "blusterIdx.z";
+    case ParallelType::CIDy:
+      return "blusterIdx.y";
+    case ParallelType::CIDx:
+      return "blusterIdx.x";
```

Comment:
Should we have two parallel types for cluster? based on our naming convention, `CIDx` means cluster in the grid, but looks like you are using it for block in the cluster? Why not have both parallel types?

Reply:
Here, I thought CIDx as grouping all the blocks in x direction into a cluster, so it is still mapped to blockIdx.x or blusterIdx.x.

Ideally, we should have TIDx for thread index in block, BIDx for block index in cluster, and CIDx for cluster index in grid. Corresponding to 3 parallel types:`ParallelType::TIDx`, `ParallelType::BIDx`, `parallelType::CIDx`. However, in the main branch, BIDx represents the block index in grid, so we need to rename the old `ParallelType::BIDx` in the main branch to maybe something like `ParallelType::BLCx` means it is a merge of **bl**ock and **c**luster? Then we can map:
```
thread index in block  --> ParallelType::TIDx --> threadIdx.x (cuda interal)
block index in cluster --> ParallelType::BIDx --> blusterIdx.x (user defined)
cluster index in grid  --> parallelType::CIDx --> clusterIdx.x  (user defined)
block index in grid    --> ParallelType::BLCx --> blockIdx.x (cuda interal)
```
Do we want this rename?

Reply:
Or maybe:

**T**hread index in **B**lock  --> ParallelType::TBIdx --> threadIdx.x (cuda interal)
**B**lock index in **C**luster --> ParallelType::BCIdx --> blusterIdx.x (user defined)
**C**luster index in **G**rid    --> parallelType::CGIdx --> clusterIdx.x  (user defined)
**B**lock index in **G**rid       --> ParallelType::BGIdx --> blockIdx.x (cuda interal)

Reply:
I prefer to keep the original `BIDx` as is. Changing it is just too big, and could be error-prone.

Reply:
> The only difficulty is just to get a good short name for "block index in cluster".

I see the difficulty. `KIDx` and `blusterIdx.x` sounds strange to me, but I don't have any better suggestions.